src/sentry/api/endpoints/organization_events.py

```diff
@@ -417,7 +420,13 @@ def _dashboards_data_fn(scopedDataset, offset, limit, scoped_query, dashboard_wi
                     has_errors = False
                     error_results = None
 
-                original_results = _data_fn(scopedDataset, offset, limit, scoped_query)
+                original_results = _data_fn(
+                    scopedDataset,
+                    offset,
+                    limit,
+                    scoped_query,
+                    fallback_to_transactions=save_discover_dataset_decision,
```

Comment:
This is the original results we return if we're unable to make a decision. For now (since we haven't decided on a default, have we?) this should be `fallback_to_transaction=False`. The transactions results below should query metricsEnhanced with `fallback_to_transaction=True`.

So now you have 3 queries:
1. Errors
2. MEP with transactions
3. MEP with discover

If both 2 and 3 return results or neither return results (ie if it's ambiguous, use the results from 3).